src/espidf.rs

```diff
@@ -360,26 +329,71 @@ impl Installer {
         log::debug!("Using PATH='{}'", &paths.to_string_lossy());
 
         Ok(EspIdf {
-            install_dir,
             esp_idf,
-            esp_idf_version: self.version,
             exported_path: paths,
-            venv_python: python,
         })
     }
 
+    /// Find a possible installed esp-idf git repository.
+    ///
+    /// This will search in te directory location supplied by the caller.
+    fn find_esp_idf(indir: impl AsRef<Path>, gitref: &git::Ref) -> Result<Option<git::Repository>> {
+        let indir = indir.as_ref();
+
+        if !indir.exists() {
+            return Ok(None);
+        }
+
+        let esp_idf_dir = indir.join(Self::esp_idf_folder_name(gitref));
+        if let Ok(repo) = git::Repository::open(&esp_idf_dir) {
+            if repo.is_ref(gitref) {
+                Ok(Some(repo))
+            } else {
+                anyhow::bail!(
+                    "Repository {} is not matching GIT ref {}",
+                    repo.worktree().display(),
+                    gitref
+                );
+            }
+        } else {
+            Ok(None)
+        }
+    }
+
     /// Clone the `esp-idf` into `repo`.
-    pub fn clone_esp_idf(&self, repo: &mut git::Repository) -> Result<()> {
+    fn clone_esp_idf(
+        url: impl AsRef<str>,
+        gitref: git::Ref,
+        repo: &mut git::Repository,
+    ) -> Result<()> {
         repo.clone_ext(
-            self.git_url
-                .as_deref()
-                .unwrap_or(DEFAULT_ESP_IDF_REPOSITORY),
-            git::CloneOptions::new()
-                .force_ref(self.version.clone())
-                .depth(1),
+            url.as_ref(),
+            git::CloneOptions::new().force_ref(gitref).depth(1),
         )?;
         Ok(())
     }
+
+    fn esp_idf_repo_name(url: impl AsRef<str>) -> String {
+        let mut sha1 = Sha1::new();
+
+        sha1.update(url.as_ref().as_bytes());
+        let bytes = sha1.finalize();
+
+        hex::encode(&bytes)
+    }
```

Comment:
I like adding a hash, but:
Instead of adding two dependencies just for this, we should instead use `std`s own has (which is only 64-bit) but should suffice. This also reduces the chars from 20->16 (cargo also has 16 chars hashes) and can be formatted just using `format!()`.

As windows has (still) a file-path length limit is think having this path 40-characters longer (as I understand it) is not that great. Though I guess we could change it if this really makes problems.

Reply:
Originally, I had big concerns with using `std::hash`, as this is a thing designed for hashmaps and hashsets, where having collisions is not ideal, but not the end of the world either. In our case, a collision should just not happen, as the consequences are pretty terrible - the wrong esp-idf will be re-used  for the build, with the user probably not even realizing that.

Now, `std::hash` seems to still be based on [SipHash](https://en.wikipedia.org/wiki/SipHash) which seems to have a strong [anti-collision](https://nnethercote.github.io/perf-book/hashing.html) guarantee but (a) is it as-strong as sha-1 or md-4 and (b) the fact that SipHash is used seems to be [now just an internal implementation detail](https://doc.rust-lang.org/std/hash/struct.SipHasher.html), so in future we might be setting ourselves for a nasty surprise, where Rust might switch to a much faster hasher, but with much weaker anti-collision guarantees, which might lead to trouble.

With all that said, *IF* cargo is using `std::hash` (where?) for a use-case similar to ours (shortening some long strings by preserving their uniqueness), I'm fine with switching to `std::hash`.

Reply:
Done. We are now using SipHash via std::hash, fingers crossed for no collisions.

Reply:
I think it will be okay, but I guess time will tell.